x-pack/plugins/observability_solution/apm/server/routes/assistant_functions/get_log_categories/index.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import datemath from '@elastic/datemath';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { getTypedSearch } from '../../diagnostics/create_typed_es_client';
+
+export type LogCategories =
+  | Array<{
+      key: string;
+      docCount: number;
+    }>
+  | undefined;
+
+export async function getLogCategories({
+  esClient,
+  arguments: args,
+}: {
+  esClient: ElasticsearchClient;
+  arguments: {
+    start: string;
+    end: string;
+  };
+}): Promise<LogCategories> {
+  const start = datemath.parse(args.start)?.valueOf()!;
+  const end = datemath.parse(args.end)?.valueOf()!;
+
+  const search = getTypedSearch(esClient);
+  const res = await search({
+    index: '*:logs-*,logs-*,*:filebeat*,filebeat*', // TODO: make this configurable somehow (UI, kibana config?)
```

Comment:
WDYT about adding this to Kibana Advanced Settings and exposing this in the AI Assistant settings?

Reply:
sgtm, for sure we need a way for it to be configurable

Reply:
WDYT about leaving this hardcoded and making it configurable in a follow-up? 
I don't mind doing it in this PR but I want to be mindful of reviewers. It's already quite big and the settings change will also require me to move the other settings to be persisted server side.

Reply:
sure, sgtm